src/Cassandra/Observers/Abstractions/IRequestObserver.cs

```diff
@@ -26,10 +27,16 @@ internal interface IRequestObserver
     {
         void OnSpeculativeExecution(Host host, long delay);
 
-        void OnRequestError(Host host, RequestErrorType errorType, RetryDecision.RetryDecisionType decision);
+        Task OnNodeStart(Host host, RequestTrackingInfo requestTrackingInfo);
 
-        void OnRequestStart();
+        Task OnNodeRequestError(Host host, RequestErrorType errorType, RetryDecision.RetryDecisionType decision, RequestTrackingInfo r, Exception ex);
 
-        void OnRequestFinish(Exception exception);
+        Task OnNodeSuccess(Host host, RequestTrackingInfo requestTrackingInfo);
+
+        Task OnRequestStart(RequestTrackingInfo requestTrackingInfo);
+
+        Task OnRequestFailure(Exception ex, RequestTrackingInfo requestTrackingInfo);
+
+        Task OnRequestSuccess(RequestTrackingInfo requestTrackingInfo);
```

Comment:
Should probably add `Async` suffix to the names of these methods to make it consistent.

Reply:
Done, please review